File: leetcode/11.scala
object Solution {
    def maxArea(height: Array[Int]): Int = {
        var most_water = 0

        var i = 0
        var j = height.length - 1

        while (j > i) {
            val left = height(i)
            val right = height(j)

            val can_hold = left.min(right) * (j - i)
            most_water = most_water.max(can_hold)

            if (left < right) {
               i += 1
            } else {
               j -= 1
            }
        }

        most_water
    }
}

File: leetcode/11.rs
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut most: i32 = 0;
        let mut i: usize = 0;
        let mut j: usize = (height.len() - 1);
        
        while j > i {
            let left: i32 = height[i];
            let right: i32 = height[j];

            let can_hold: i32 = left.min(right) * (j - i) as i32; 
            most = most.max(can_hold);

            if left < right {
               i += 1;
            } else {
               j -= 1;
            }

        }

        most
    }
}